android/envoy/src/main/kotlin/org/greatfire/envoy/NetworkIntentService.kt

```diff
@@ -759,6 +769,74 @@ class NetworkIntentService : IntentService("NetworkIntentService") {
         }
     }
 
+    private fun handleMeekSubmit(
+        url: String,
+        captive_portal_url: String,
+        hysteriaCert: String?
+    ) {
+
+        val uri = URI(url)
+        var meekUrl = ""
+        var meekFront = ""
+        var meekTunnel = ""
+        val rawQuery = uri.rawQuery
+        val queries = rawQuery.split("&")
+        for (i in 0 until queries.size) {
+            val queryParts = queries[i].split("=")
+            if (queryParts[0].equals("url")) {
+                meekUrl = URLDecoder.decode(queryParts[1], "UTF-8")
+            } else if (queryParts[0].equals("front")) {
+                meekFront = URLDecoder.decode(queryParts[1], "UTF-8")
+                // if needed, generate randomized host name prefix
+                if (meekFront.startsWith('.')) {
+                    meekFront = randomString().plus(meekFront)
+                    Log.d(TAG, "front updated with random prefix: " + meekFront)
+                } else {
+                    Log.d(TAG, "front included as-is: " + meekFront)
+                }
+            } else if (queryParts[0].equals("tunnel")) {
+                // this is purposfully not decocded to add to an Envoy URL
+                meekTunnel = queryParts[1]
+            }
+        }
+        // start meek service
+        if (meekUrl.isNullOrEmpty() || meekFront.isNullOrEmpty() || meekTunnel.isNullOrEmpty()) {
+            Log.e(TAG, "some arguments required for meek service are missing")
+        } else {
+
+            // set the login params
+            val meekUserString = "url=" + meekUrl + ";front=" + meekFront
+            val nullCharString = "" + Char.MIN_VALUE
+
+            // additional hardcoded meek parameters
+            val logLevel = "DEBUG"
```

Comment:
why is debug / unsafeLogging hard coded to on?

Reply:
fixed.  now set to ERROR/false/false